Classes/iOSDeviceManagerLogging.m

```diff
@@ -0,0 +1,230 @@
+
+#ifndef DD_LEGACY_MACROS
+#define DD_LEGACY_MACROS 0
+#endif
+
+#import "iOSDeviceManagerLogging.h"
+#import "DDTTYLogger.h"
+#import "DDASLLogger.h"
+#import "DDFileLogger.h"
+#import "DDLog.h"
+#import <libkern/OSAtomic.h>
+
+#pragma mark - IDMLogFileManager
+
+static NSString *const IDMLogFileNameDateFormat = @"yyyy-MM-dd-HH-mm-ss";
+static NSString *const IDMLogFileNameDateFormatterKey = @"sh.calaba-IDMLogFileNameFormatter-NSDateFormatter";
+static NSString *const IDMLogFilePrefix = @"idm-";
+
+@interface DDLogFileManagerDefault (iOSDeviceManagerAdditions)
+
+- (void)deleteOldLogFiles;
+
+@end
+
+@interface IDMLogFileManager : DDLogFileManagerDefault
+
+@end
+
+@implementation IDMLogFileManager
+
+
+- (NSString *)newLogFileName {
+
+  NSDateFormatter *dateFormatter = [self logFileDateFormatter];
+  NSString *formattedDate = [dateFormatter stringFromDate:[NSDate date]];
+
+  return [NSString stringWithFormat:@"%@%@.log", IDMLogFilePrefix, formattedDate];
+}
+
+
+- (NSDateFormatter *)logFileDateFormatter {
+  NSMutableDictionary *dictionary = [[NSThread currentThread] threadDictionary];
+  NSString *dateFormat = IDMLogFileNameDateFormat;
+  NSDateFormatter *dateFormatter = dictionary[IDMLogFileNameDateFormatterKey];
+
+  if (dateFormatter == nil) {
+    dateFormatter = [[NSDateFormatter alloc] init];
+    [dateFormatter setLocale:[NSLocale localeWithLocaleIdentifier:@"en_US_POSIX"]];
+    [dateFormatter setDateFormat:dateFormat];
+    [dateFormatter setTimeZone:[NSTimeZone timeZoneForSecondsFromGMT:0]];
+    dictionary[IDMLogFileNameDateFormatterKey] = dateFormatter;
+  }
+
+  return dateFormatter;
+}
+
+- (NSString *)createNewLogFile {
+  NSString *fileName = [self newLogFileName];
+  NSString *logsDirectory = [self logsDirectory];
```

Comment:
The logsDirectory property is set in the designated initializer.

This is a subclass of the default DD log file manager.